pkl-executor/src/main/java/org/pkl/executor/ExecutorException.java

```diff
@@ -19,11 +19,22 @@
  * Indicates an {@link Executor} error. {@link #getMessage()} returns a user-facing error message.
  */
 public final class ExecutorException extends RuntimeException {
+  String pklVersion = "unresolved";
+
   public ExecutorException(String message) {
     super(message);
   }
 
   public ExecutorException(String message, Throwable cause) {
     super(message, cause);
   }
+
+  public ExecutorException(String message, String version, Throwable cause) {
+    super(message, cause);
+    pklVersion = version;
+  }
+
+  public String getPklVersion() {
+    return pklVersion;
+  }
```

Comment:
Some nits:

* We can probably just return `null` if there is no pkl version to relate.
* Add a doc comment to `getPklVersion()`
* Would be nice to add the Pkl version to `getMessage()`

```suggestion
  String pklVersion = null;

  public ExecutorException(String message) {
    super(message);
  }

  public ExecutorException(String message, Throwable cause) {
    super(message, cause);
  }

  public ExecutorException(String message, String version, Throwable cause) {
    super(message, cause);
    pklVersion = version;
  }

  /**
   * The selected Pkl version used to evaluate the module.
   *
   * Returns {@code null} if this exception does not originate from an underlying Pkl evaluator.
   */
  public String getPklVersion() {
    return pklVersion;
  }

  @Override
  public String getMessage() {
    var message = super.getMessage();
    if (pklVersion == null) {
      return message;
    }
    return message + "\nPkl version: " + pklVersion;
  }
```

Reply:
* Should be `private final String pklVersion`.
* New constructor should append parameter: `public ExecutorException(String message, Throwable cause, String version)`